packages/shared/lib/validateAddress.tsx

```diff
@@ -0,0 +1,29 @@
+import * as WAValidator from 'multicoin-address-validator';
+export class AddressValidator {
+  public isValidAddress(address: string, networkProtocol: string | undefined): boolean {
+    if (this.doesValidatorExist(networkProtocol)) {
+      const isValid = WAValidator.validate(address, networkProtocol);
+      return isValid;
+    } else {
+      return this.customValidator(address, networkProtocol);
+    }
+  }
+
+  private doesValidatorExist(networkProtocol: string | undefined): boolean {
+    return !!networkProtocol && !!WAValidator.findCurrency(networkProtocol);
```

Comment:
not sure if we need a null check for networkProtocol or not. Do we have test cases for the functions?

Reply:
@dainkang1 this is a new function so no.
It seems like the only possible way to get this is by getting an asset which we don't know exists, which I think is redundant (since in such cases we don't provide a means to derive it using the UI).

Let's leave this as is as a security measure, since eventually customers might alter the code and cause issues by missing properties.

@ArielSir-fb I think in the event that a networkProtocol is undefined we should either log an error so that customers will know in the case of missing protocol on a new asset they've added (or we will know).
This relates more to adding the logs we discussed, so it should be added as part of it.

Reply:
@dainkang1 This is actually a precaution when using the external lib for address validation. So far the only chain I noticed not supported by this lib is HBAR, hence I prepared a "safety net" where we can add custom validations like I did HBAR if required.

@agoltzman I will add logging and check tests